mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/search/ValueMatcher.java

```diff
@@ -0,0 +1,98 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault.search;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Locale;
+import java.util.function.BiFunction;
+
+public interface ValueMatcher<T> {
+
+    boolean matches(T referenceValue);
+
+    class SingleValueMatcher<T> implements ValueMatcher<T> {
+
+        private static final BiFunction<String, String, Boolean> CONTAINS = (testedValue, referenceValue) -> referenceValue.contains(testedValue);
+        private static final BiFunction<String, String, Boolean> CONTAINS_IGNORE_CASE =
+            (testedValue, referenceValue) -> referenceValue.toLowerCase(Locale.US).contains(testedValue.toLowerCase(Locale.US));
```

Comment:
You can't choose an arbitrary Locale when you match user data because you don't know yet the right Locale you have to pick.

Reply:
Well, we clearly don't have better solution here.

Maybe the locale of the system would be slightly better.

Do you have better suggestion @mbaechler  ?

Reply:
Okay, the final decision is keeping this usage of locale and I will document the warning about locale in webadmin markdown in latter PR of Vault API.

Reply:
> Okay, the final decision is keeping this usage of locale and I will document the warning about locale in webadmin markdown in latter PR of Vault API.

It's not what I suggested (and agreed): I asked to write a failing test, open a JIRA for documentation and disable that test with the JIRA number.
The webadmin documentation is nice too.

Reply:
it is arrived: 99fb4fd